packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -442,7 +442,8 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
   useEffect(() => {
     // Reset focus path
     setFocusPath(params.path ? pathFromString(params.path) : [])
-  }, [documentId, params.path])
+    onSetOpenPath([])
+  }, [params.path, schemaType?.fields, documentId])
```

Comment:
Unless I'm missing something, `schemaType?.fields` is not needed as a dependency here?
```suggestion
  }, [params.path, documentId])
```